### DISCUSSION TITLE
Integrated Second/Third Party Polyfill Instalation

### DISCUSSION THREAD

**Discussion by jcc10** [denoland/deno#19594] [Ideas]
This is more or less a Request For Comment on if this should be made into a actual issue or if it is just stupid, feel free to tear it apart, just please be gentle ok?

There are a number of standards where it is difficult to integrate into deno directly due to problems testing for example:

* WebUSB
* WebAuthn
* Web Bluetooth
* WebHID
* And a lot more...

The common factor for all of these is that it requires hardware testing and integration. I would like to state that I completely understand why adding these is difficult, time-consuming, and low priority as they are niche use cases.

I feel like it might be nice to add some way to install Second or Third Party plugins to polyfill the functionality so you don't have to enable FFI for a entire script instead installing a plugin that register's that it can fill the requirement itself that gets FFI access, and not propagating said access to the rest of the script.

Issues I can see with adding this:

- How to allow for registering permissions to be piped to the plugins. (Via command line as interactive prompts would be easier)
- Syntax for registering it's object.
- How much overhead would said plugin system add?
- Would we limit the plugin system to only official API's or would we allow creation of new API's that would be commonly requested. (I.E. A WebView API)
- How would plugin's like this get compiled? Would it even be possible?
- Would all modules be Third Party? Would there eventually be a list of "Second Party" plugins (plugins that get some kind of vetting by the deno community?) Would there eventually be a list of First Party plugins that are run as community groups to add functionality?

There are probably more, but these are the ones I could think of off the top of my head though I'm sure there is more.

Tl;Dr: I want API's that would require FFI to polyfill, would like a way to give FFI to one small, verifiable script and allow every other script to call through the small script, thereby not giving FFI to every script that would need the API's.
